src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/HMACMD5.cs

```diff
@@ -155,6 +158,181 @@ public static bool TryHashData(ReadOnlySpan<byte> key, ReadOnlySpan<byte> source
+        /// </summary>
+        /// <param name="key">The HMAC key.</param>
+        /// <param name="source">The stream to HMAC.</param>
+        /// <returns>The HMAC of the data.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="source" /> does not support reading.
+        /// </exception>
+        public static byte[] HashData(ReadOnlySpan<byte> key, Stream source)
+        {
+            ArgumentNullException.ThrowIfNull(source);
+
+            if (!source.CanRead)
+                throw new ArgumentException(SR.Argument_StreamNotReadable, nameof(source));
+
+            return LiteHashProvider.HmacStream(HashAlgorithmNames.MD5, HmacSizeBytes, key, source);
+        }
+
+        /// <summary>
+        /// Computes the HMAC of a stream using the MD5 algorithm.
+        /// </summary>
+        /// <param name="key">The HMAC key.</param>
+        /// <param name="source">The stream to HMAC.</param>
+        /// <returns>The HMAC of the data.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="key" /> or <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="source" /> does not support reading.
+        /// </exception>
+        public static byte[] HashData(byte[] key, Stream source)
+        {
+            ArgumentNullException.ThrowIfNull(key);
+            return HashData(new ReadOnlySpan<byte>(key), source);
+        }
+
+        /// <summary>
+        /// Asynchronously computes the HMAC of a stream using the MD5 algorithm.
+        /// </summary>
+        /// <param name="key">The HMAC key.</param>
+        /// <param name="source">The stream to HMAC.</param>
+        /// <param name="cancellationToken">
+        ///   The token to monitor for cancellation requests.
+        ///   The default value is <see cref="System.Threading.CancellationToken.None" />.
+        /// </param>
+        /// <returns>The HMAC of the data.</returns>
+        /// <exception cref="ArgumentNullException">
+        ///   <paramref name="source" /> is <see langword="null" />.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        ///   <paramref name="source" /> does not support reading.
+        /// </exception>
+        public static ValueTask<byte[]> HashDataAsync(ReadOnlyMemory<byte> key, Stream source, CancellationToken cancellationToken = default)
+        {
+            if (source is null)
+                return ValueTask.FromException<byte[]>(new ArgumentNullException(nameof(source)));
+
+            if (!source.CanRead)
+                return ValueTask.FromException<byte[]>(
+                    new ArgumentException(SR.Argument_StreamNotReadable, nameof(source)));
```

Comment:
We typically don't wrap user errors (e.g. argument exceptions) in tasks.

Reply:
Ha, that makes sense. Fixed.